web/clusters.go

```diff
@@ -11,6 +12,182 @@ import (
 	"github.com/trento-project/trento/internal/hosts"
 )
 
+type Node struct {
+	Name       string
+	Attributes map[string]string
+	Resources  []*Resource
+	Ip         string
+	Health     string
+	VirtualIps []string
+}
+
+type Resource struct {
+	Id        string
+	Type      string
+	Role      string
+	Status    string
+	FailCount int
+}
+
+type Nodes []*Node
+
+func stoppedResources(c *cluster.Cluster) []*Resource {
+	var stoppedResources []*Resource
+
+	for _, r := range c.Crmmon.Resources {
+		if r.NodesRunningOn == 0 && !r.Active {
+			resource := &Resource{
+				Id: r.Id,
+			}
+			stoppedResources = append(stoppedResources, resource)
+		}
+	}
+
+	return stoppedResources
+}
+
+func (node *Node) Role() string {
+	if r, ok := node.Attributes["hana_prd_roles"]; ok {
+		role := r[strings.LastIndex(r, ":")+1:]
+		return strings.Title(role)
+	}
+	return "-"
+}
+
+func (node *Node) HealthState() string {
+	if r, ok := node.Attributes["hana_prd_roles"]; ok {
+		healthState := strings.SplitN(r, ":", 2)[0]
+		return healthState
+	}
+	return "-"
+}
+
+func (node *Node) Status() string {
+	if r, ok := node.Attributes["hana_prd_roles"]; ok {
+		status := strings.SplitN(r, ":", 3)[1]
+
+		switch status {
+		case "P":
+			return "Primary"
+		case "S":
+			return "Secondary"
+		}
+	}
+	return "-"
+}
+
+func NewNodes(c *cluster.Cluster, hl hosts.HostList) Nodes {
+	var nodes Nodes
+
+	for _, n := range c.Crmmon.NodeAttributes.Nodes {
+		node := &Node{Name: n.Name, Attributes: make(map[string]string)}
+
+		for _, a := range n.Attributes {
+			node.Attributes[a.Name] = a.Value
+		}
+
+		for _, r := range c.Crmmon.Resources {
```

Comment:
I think we need to loop through the `Groups` and `Clones` too. Individual resources are just one part of the resources.